OutSystems/outsystems-maps
# Main

This Repository is for...

Google Maps Prototype: As I am a contributor in this app, so I wanted this to be one of my repository
This module will give you clear understanding of how to use Google Map Mobile plugin with Direction enabled feature and use of carosel with map widget with respect to markers Note : For enabling direction feature we will require API Key.
[https://www.outsystems.com/forge/component-overview/12226/google-map-model-prototype-app]

The challenge which we faced is not getting location after adding carousel on map and changing the position on change of carousel. So we decided to build a demo application with the above working functionality

Test Steps
![1641208968925](https://user-images.githubusercontent.com/52080797/185338380-a5097964-314f-4c65-afc8-486dbe8fd559.png)
1. 1. Using the Location plugin, in client action, we can get location. Then append the list into the map marker list.
2. 
![1641214461316](https://user-images.githubusercontent.com/52080797/185338433-4a4c155e-1d53-4ce4-833e-db56591a07c8.png)
In this client action, we clear the marker list
Then we store the marker data in the map marker variable.
Then we append the map marker variable to the map marker list.
i.e. using the MapMarker list variable we can show all markers on the map.
3.
![1641214551572](https://user-images.githubusercontent.com/52080797/185338984-e6d9805b-e71c-4dc2-864a-3fa81b391891.png)
This event is used for displaying the information related to the marker when clicked.
-Referred from GoogleMapLibrary
Current Location on click  – This will show you the current location of the user when clicked on the icon
4.![1641214661891](https://user-images.githubusercontent.com/52080797/185339045-8958977f-2d89-4703-856c-865d0fd4765b.png)
In this client action, we can get the live location of the user with all the markers on the map. So, in this action, we can assign the current lat and long to the coordinate variable,
Then we reset the direction to disable the direction feature.
5.![1641214773904](https://user-images.githubusercontent.com/52080797/185339162-5efa1416-8133-47db-b027-28d10492d430.png)
This action will perform when the carousel is changed
check index >=0

1.a. If the condition is false then the action will end and the condition and,
1.b. If it is true then the marker will be set to the center location.
2. Then we assign map direction lat & long to origin and destination which will enable direction.
check showDirection condition
3.a. if it is false then action will end
3.b. if the condition is true then the markers list will clear then append that current marker in the marker list variable. And action will end. 

6.![1641214816850](https://user-images.githubusercontent.com/52080797/185339438-1d0ddd2d-653e-4956-8d19-d36fbbd85b8a.png)
Start, the Current location of the user will be fetched.
Assign, Coordinate Lat & Long – For the center position of the marker on the screen.
GoogleMap_CircleCreate, CircleObj – Will create a ring around your current location as shown in the image.

7.![1641214850542](https://user-images.githubusercontent.com/52080797/185339538-cbff11ee-572d-4eed-997f-a1196f60e1eb.png)








### Screenshots
(prefer animated gif)


### Checklist
